pkg/api/router.go

```diff
@@ -105,10 +105,18 @@ func authHandler(cfg *Config, handler http.HandlerFunc) http.HandlerFunc {
 
 	if cfg.Auth.BearerToken != "" {
 		return func(w http.ResponseWriter, r *http.Request) {
+			errLogs := func(token string) {
+				log.Error("msg", "Unauthorized access to endpoint, invalid bearer token: "+token)
+				http.Error(w, "Unauthorized access to endpoint, invalid bearer token.", http.StatusUnauthorized)
+			}
 			splitToken := strings.Split(r.Header.Get("Authorization"), "Bearer ")
+			if len(splitToken) < 2 {
+				// todo: add test for this.
+				errLogs("")
+				return
+			}
```

Comment:
This is no longer part of this PR. Once the conflicts are resolved, this would be done.